packages/react/src/components/ButtonGroup/ButtonGroup.js

```diff
@@ -16,26 +16,115 @@ const { prefix } = settings;
 /**
  * Button group.
  */
-const ButtonGroup = ({ buttons }) => (
-  <ol
-    className={`${prefix}--buttongroup`}
-    data-autoid={`${stablePrefix}--button-group`}>
-    {buttons.map((button, key) => {
-      return (
-        <li key={key} className={`${prefix}--buttongroup-item`}>
-          <Button
-            tabIndex={key === 0 ? 2 : 1}
-            data-autoid={`${stablePrefix}--button-group-${key}`}
-            {...button}
-            type="button"
-            kind={key === 0 ? 'primary' : 'tertiary'}>
-            {button.copy}
-          </Button>
-        </li>
+const ButtonGroup = ({ buttons, enableSizeByContent }) => {
+  const groupRef = useRef(null);
+  const observedPseudoButtonNodesRef = useRef(new Set());
+  const shouldUseResizeObserver =
+    enableSizeByContent && typeof ResizeObserver !== 'undefined';
+  const resizeObserverButtonsRef = useRef(
+    new ResizeObserver(entries => {
```

Comment:
```suggestion
    !shouldUseResizeObserver  ? null : new ResizeObserver(entries => {
```

My apologies that I may have missed this in my prototype. This may solve CI error.